backend/src/services/auto_trader.py

```
@@ -33,238 +23,382 @@ def __init__(self, socketio=None, data_fetcher_service=None, ai_analysis_service
+            if response.status_code == 200:
+                rug_report = response.json()
+                score = rug_report.get('score', 0)
+                max_allowed_score = self.config.get("rugcheck_max_score", 5000)
+
+                if score > max_allowed_score:
+                    logger.warning(f"AutoTrader: Skipping {token.get('symbol', token_address)} - RugCheck score ({score}) exceeds max allowed ({max_allowed_score}).")
+                    return
+                logger.info(f"AutoTrader: RugCheck passed for {token_address} with score: {score}")
+            else:
+                logger.warning(f"AutoTrader: RugCheck API returned status {response.status_code}. Proceeding with caution.")
+        except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)